distributed/scheduler.py

```diff
@@ -503,9 +503,18 @@ def cleanup(self):
         self.status = 'closing'
         logger.debug("Cleaning up coroutines")
 
-        for w, comm in list(self.worker_comms.items()):
-            with ignoring(AttributeError):
-                yield comm.close()
+        close_nannies = {w: self.rpc(self.get_worker_service_addr(w, 'nanny')).terminate()
```

Comment:
Unrelated to this PR, but I'm wondering why the scheduler has to close both workers and nannies. Is it a historical accident or a deliberate decision? Intuitively, closing e.g. the worker to close explicitly should also tell the nanny to exit (or the reverse).